src/Popover/Popover.tsx

```diff
@@ -321,23 +392,23 @@ export default class Popover extends React.Component {
     }
   };
 
-  _onMouseEvent = (type) => {
+  _onMouseEvent = (type: string) => {
     if (type === 'mouseenter') {
       this.showPopover();
     } else if (type === 'mouseleave') {
       this.hidePopover();
     }
   };
 
-  onMouseEvent = ({ type }) => {
+  onMouseEvent: React.MouseEventHandler<any> = ({ type }) => {
     const { delay } = this.getPopoverProps();
     const delayWhenClosed = this.getDelayWhenClosed(delay);
     const delayWhenOpen = this.getDelayWhenOpen(delay);
 
     if (this.isOpen()) {
-      return delayWhenOpen ? this.onMouseEventDebouncedWhenOpen(type) : this._onMouseEvent(type);
+      return delayWhenOpen ? this.onMouseEventDebouncedWhenOpen && this.onMouseEventDebouncedWhenOpen(type) : this._onMouseEvent(type);
```

Comment:
very naive way of handling this, but I don't have better options. I didn't find any guarantee that `onMouseEventDebouncedWhenOpen` is never null.

Reply:
it's enforced by `updateDebouncedMousedEvents`  in `componentWillReceiveProps`
anyway, no problem in double checking here

(this flow is complex and confusing, I know)